### PR TITLE
Chore/3.12.0

### PR DESCRIPTION
**Pull request by picodoth**
First of all, thank you for your contribution! 😄

Please makes sure that these form are filled before submitting your pull request, thank you!

[[中文版](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / document update
- [ ] Branch merge
- [ ] Other (about what?)

### What's the background?
  
> 1. Describe the source of requirement.
> 2. Resolve what problem.
> 3. Related issue link.
  
### API Realization (Optional if not new feature)
  
> 1. Basic thought of solution and other optional proposal.
> 2. List final API realization and usage sample.
> 3. GIF or snapshot should be provided if includes UI/interactive modification.
  
### What's the effect? (Optional if not new feature)

> 1. Does this PR affect user? Which part will be affected?
> 2. What will say in changelog?
> 3. Does this PR contains potential break change or other risk?

### Self Check before Merge

- [ ] Doc is updated/provided or not needed
- [ ] Demo is updated/provided or not needed
- [ ] TypeScript definition is updated/provided or not needed
- [ ] Changelog provided

### Additional Plan? (Optional if not new feature)

> If this PR related with other PR or following info. You can type here.

### CODE REVIEW

File: CHANGELOG.en-US.md [hunk 1]
```diff
@@ -15,6 +15,71 @@ timeline: true
 
 ---
 
+## 3.12.0
+
+- 🎉 New component [Empty](https://ant.design/components/empty/)! [13651](https://github.com/ant-design/ant-design/pull/13651)
+- 🎉 Kannada locale added (kn_IN). [#13776](https://github.com/ant-design/ant-design/pull/13776) [@gurungrahul2](https://github.com/gurungrahul2)
+- 🌟 Add `eye-invisible` Icon. [b8630b3](https://github.com/ant-design/ant-design/commit/b8630b34556e58af31f51fb42d1299725ddd8219)
+- 🌟 ConfigProvider support prefixCls. [#13389](https://github.com/ant-design/ant-design/pull/13389)
```
Comment (ycjcl868) [inline CHANGELOG.en-US.md]: ConfigProvider component

File: CHANGELOG.en-US.md [hunk 2]
```diff
@@ -15,6 +15,71 @@ timeline: true
...
+  - 🐞 Fixed disabled button style in DatePicker panel. [#14098](https://github.com/ant-design/ant-design/pull/14098)
+- Calendar
+  - 🌟 Calendar support prop `renderExtraFooter` in all mode. [#13813](https://github.com/ant-design/ant-design/pull/13813) [@elfman](https://github.com/elfman
+  - 🐞 Fixed month range display when start year equals end year. [#14049](https://github.com/ant-design/ant-design/pull/14049) [@meihuanyu](https://github.com/meihuanyu)
+- TimePicker
+  - 🌟 TimePicker support new prop `popupStyle` and `onAmPmChange`. [833c181](https://github.com/ant-design/ant-design/commit/833c18192247f265b4004afa11a054846c7ba662
+  - 🐞 Fixed TimePicker Icon disappear when used with Input.Group. [#13797](https://github.com/ant-design/ant-design/pull/13797) [@mraiguo](https://github.com/ant-design/ant-design/pull/13797)
+- 🌟 Update `rc-tree-select` to 2.5.0 to support use a ReactNode as `notFoundContent` in TreeSelect. [47b89e5](https://github.com/ant-design/ant-design/commit/47b89e56fbedfa07a4c263ca390a78d58132563f)
+- 🌟 Rate support `tooltips`. [192e188](https://github.com/ant-design/ant-design/commit/192e188fe87018ad9d004b9c2002e2b0045fa4b4)
```
Comment (ycjcl868) [inline CHANGELOG.en-US.md]: Rate component

File: CHANGELOG.zh-CN.md [hunk 3]
```diff
@@ -15,6 +15,69 @@ timeline: true
...
+  - 🌟 Dropdown `overlay` 支持回调函数。[#14003](https://github.com/ant-design/ant-design/pull/14003)
+- Button
+  - 🐞 修复 Button 使用自定义类型抛出时的问题。[#13915](https://github.com/ant-design/ant-design/issues/13915)
+  - 🐞 修复按钮中有 plus 和 minus 图标时在 Window 下的样式问题。[#13924](https://github.com/ant-design/ant-design/issues/13924)
+- Table
+  - 🌟 添加属性 `sortDirections` 到 Table 和 Table.Column。[#13773](https://github.com/ant-design/ant-design/pull/13773) [@elfman](https://github.com/elfman)
+  - 🐞 修复 Badge 组件遮盖 Table 固定列的问题。[#13930](https://github.com/ant-design/ant-design/issues/13930) 
+  - 🐞 修复行选择器列的 `columnWidth` 设置不生效的问题。[#14115](https://github.com/ant-design/ant-design/pull/14115) [@dyygtfx](https://github.com/dyygtfx)
+- DatePicker
+  - 🌟 添加新的`renderFooter` API 到 DatePicker. [3c007a8](https://github.com/ant-design/ant-design/commit/3c007a85dd8a670f66c7e0aed95d3537e25ef6ea)
```
Comment (chenshuai2144) [inline CHANGELOG.zh-CN.md]: DatePicker 添加 renderFooter api？

File: CHANGELOG.zh-CN.md [hunk 4]
```diff
@@ -15,6 +15,69 @@ timeline: true
...
+  - 🐞 修复 WeekPicker 不支持 dateRender 的问题。[#13957](https://github.com/ant-design/ant-design/issues/13957)
+  - 🐞 修复禁用按钮在 DatePicker 面板中的样式问题。[#14098](https://github.com/ant-design/ant-design/pull/14098)
+- Calendar
+  - 🌟 Calendar 在所有模式中支持 `renderExtraFooter`。[#13813](https://github.com/ant-design/ant-design/pull/13813) [@elfman](https://github.com/elfman
+  - 🐞 修复月份选择器在开始年份和结束年份相等时的显示问题。 [#14049](https://github.com/ant-design/ant-design/pull/14049) [@meihuanyu](https://github.com/meihuanyu)
+- TimePicker
+  - 🌟 TimePicker 支持新的属性 `popupStyle` 和 `onAmPmChange`。[833c181](https://github.com/ant-design/ant-design/commit/833c18192247f265b4004afa11a054846c7ba662
```
Comment (chenshuai2144) [inline CHANGELOG.zh-CN.md]: 添加新的属性？

File: CHANGELOG.zh-CN.md [hunk 5]
```diff
@@ -15,6 +15,69 @@ timeline: true
...
+  - 🐞 修复 TimePicker 在跟 Input.Group 一起使用时图标会消失的问题。[#13797](https://github.com/ant-design/ant-design/pull/13797) [@mraiguo](https://github.com/ant-design/ant-design/pull/13797)
+- 🌟 更新 `rc-tree-select` 到 2.5.0 支持 TreeSelect 中的 `notFoundContent` 可以是 ReactNode。[47b89e5](https://github.com/ant-design/ant-design/commit/47b89e56fbedfa07a4c263ca390a78d58132563f)
```
Comment (chenshuai2144) [inline CHANGELOG.zh-CN.md]: TreeSelect 中的 `notFoundContent`  支持传入 ReactNode

File: CHANGELOG.zh-CN.md [hunk 6]
```diff
@@ -15,6 +15,69 @@ timeline: true
...
+  - 🐞 修复 Popover 上方的 border 在某些 IE9 下不显示的问题。[#14064](https://github.com/ant-design/ant-design/issues/14064) [@gyh9457](https://github.com/gyh9457)
+- 🐞 修复 Radio 组件在 Chrome 下的样式问题。[#3699](https://github.com/ant-design/ant-design/issues/3699)
+- 🐞 修复 Steps 组件在 IE9 下的样式问题。[#14001](https://github.com/ant-design/ant-design/issues/14001)
+- 🐞 修复嵌套的 TimeLine 最后一条线丢失的问题。 [#14108](https://github.com/ant-design/ant-design/pull/14110)
+- 🐞 修复 Spin 组件初始设置 `delay` 属性后不显示的问题。[#14100](https://github.com/ant-design/ant-design/issues/14100)
+- Badge
+  - 🐞 修复当 `count` 是一个 ReactNode 时动画跳动的问题。[#13800](https://github.com/ant-design/ant-design/issues/13800)
+- TypeScript
+  - 🐞 修复 DatePicker 属性 `disabledDate` 的定义问题。[#14008](https://github.com/ant-design/ant-design/pull/14008) [@vnguyen94](https://github.com/vnguyen94)
+  - 🐞 修复 Dropdown 属性的 `onVisibleChange`  的定位问题。[#13988](https://github.com/ant-design/ant-design/pull/13988) [@travikk](https://github.com/travikk)
```
Comment (chenshuai2144) [inline CHANGELOG.zh-CN.md]: 定义问题？

File: CHANGELOG.zh-CN.md [hunk 7]
```diff
@@ -15,6 +15,69 @@ timeline: true
...
+- TimePicker
+  - 🌟 TimePicker 支持新的属性 `popupStyle` 和 `onAmPmChange`。[833c181](https://github.com/ant-design/ant-design/commit/833c18192247f265b4004afa11a054846c7ba662
+  - 🐞 修复 TimePicker 在跟 Input.Group 一起使用时图标会消失的问题。[#13797](https://github.com/ant-design/ant-design/pull/13797) [@mraiguo](https://github.com/ant-design/ant-design/pull/13797)
+- 🌟 更新 `rc-tree-select` 到 2.5.0 支持 TreeSelect 中的 `notFoundContent` 可以是 ReactNode。[47b89e5](https://github.com/ant-design/ant-design/commit/47b89e56fbedfa07a4c263ca390a78d58132563f)
+- 🌟 组件 Rate 支持 `tooltips`。[192e188](https://github.com/ant-design/ant-design/commit/192e188fe87018ad9d004b9c2002e2b0045fa4b4)
+- 🐞 修复 Drawer 的属性 `style` 不生效的问题。[#13850](https://github.com/ant-design/ant-design/issues/13850)
+- Upload
+  - 💄 添加新的 Less 变量 `upload-picture-card-border-style` 并修复 `upload-picture-card-size` 的拼写错误。[#13919](https://github.com/ant-design/ant-design/pull/13919) [#13929](https://github.com/ant-design/ant-design/pull/13929) [@flexchen](https://github.com/flexchen)
+  - 🐞 修复在 Upload 组件中无法识别 `dpg` 后缀文件为图片的问题。[#13988]
```
Comment (chenshuai2144) [inline CHANGELOG.zh-CN.md]: url消失了
修复在 Upload 组件中无法识别后缀为 `dpg`  的图片问题。

File: CHANGELOG.en-US.md [hunk 8]
```diff
@@ -15,6 +15,71 @@ timeline: true
...
+- Table
+  - 🌟 Add prop `sortDirections` for Table and Table.Column. [#13773](https://github.com/ant-design/ant-design/pull/13773) [@elfman](https://github.com/elfman)
+  - 🐞 Fixed Badge component display over the fixed table column. [#13930](https://github.com/ant-design/ant-design/issues/13930) 
+  - 🐞 Fixed rowSelection columnWidth doesn't work. [#14115](https://github.com/ant-design/ant-design/pull/14115) [@dyygtfx](https://github.com/dyygtfx)
+- DatePicker
+  - 🌟 Add new `renderFooter` API for DatePicker. [3c007a8](https://github.com/ant-design/ant-design/commit/3c007a85dd8a670f66c7e0aed95d3537e25ef6ea)
+  - 🐞 Fixed dateRender not supported at WeekPicker. [#13957](https://github.com/ant-design/ant-design/issues/13957)
+  - 🐞 Fixed disabled button style in DatePicker panel. [#14098](https://github.com/ant-design/ant-design/pull/14098)
+- Calendar
+  - 🌟 Calendar support prop `renderExtraFooter` in all mode. [#13813](https://github.com/ant-design/ant-design/pull/13813) [@elfman](https://github.com/elfman
```
Comment (chenshuai2144) [inline CHANGELOG.en-US.md]: 少了个括号？

File: CHANGELOG.en-US.md [hunk 9]
```diff
@@ -15,6 +15,70 @@ timeline: true
...
```
Comment (ycjcl868) [inline CHANGELOG.en-US.md]: lack a `)`

File: CHANGELOG.en-US.md [hunk 10]
```diff
@@ -15,6 +15,71 @@ timeline: true
...
+- TimePicker
+  - 🌟 TimePicker support new prop `popupStyle` and `onAmPmChange`. [833c181](https://github.com/ant-design/ant-design/commit/833c18192247f265b4004afa11a054846c7ba662
+  - 🐞 Fixed TimePicker Icon disappear when used with Input.Group. [#13797](https://github.com/ant-design/ant-design/pull/13797) [@mraiguo](https://github.com/ant-design/ant-design/pull/13797)
+- 🌟 Update `rc-tree-select` to 2.5.0 to support use a ReactNode as `notFoundContent` in TreeSelect. [47b89e5](https://github.com/ant-design/ant-design/commit/47b89e56fbedfa07a4c263ca390a78d58132563f)
+- 🌟 Rate support `tooltips`. [192e188](https://github.com/ant-design/ant-design/commit/192e188fe87018ad9d004b9c2002e2b0045fa4b4)
+- 🐞 Fixed Drawer prop `style` not working. [#13850](https://github.com/ant-design/ant-design/issues/13850)
+- Upload
+  - 💄 Add new less var `upload-picture-card-border-style` and Fixed `upload-picture-card-size` typo. [#13919](https://github.com/ant-design/ant-design/pull/13919) [#13929](https://github.com/ant-design/ant-design/pull/13929) [@flexchen](https://github.com/flexchen)
+  - 🐞 Fixed recognize `dpg` file type as image in Upload. [#13988]
```
Comment (chenshuai2144) [inline CHANGELOG.en-US.md]: url不见了

File: CHANGELOG.zh-CN.md [hunk 11]
```diff
@@ -15,6 +15,69 @@ timeline: true
...
+- Table
+  - 🌟 添加属性 `sortDirections` 到 Table 和 Table.Column。[#13773](https://github.com/ant-design/ant-design/pull/13773) [@elfman](https://github.com/elfman)
+  - 🐞 修复 Badge 组件遮盖 Table 固定列的问题。[#13930](https://github.com/ant-design/ant-design/issues/13930) 
+  - 🐞 修复行选择器列的 `columnWidth` 设置不生效的问题。[#14115](https://github.com/ant-design/ant-design/pull/14115) [@dyygtfx](https://github.com/dyygtfx)
+- DatePicker
+  - 🌟 添加新的`renderFooter` API 到 DatePicker. [3c007a8](https://github.com/ant-design/ant-design/commit/3c007a85dd8a670f66c7e0aed95d3537e25ef6ea)
+  - 🐞 修复 WeekPicker 不支持 dateRender 的问题。[#13957](https://github.com/ant-design/ant-design/issues/13957)
+  - 🐞 修复禁用按钮在 DatePicker 面板中的样式问题。[#14098](https://github.com/ant-design/ant-design/pull/14098)
+- Calendar
+  - 🌟 Calendar 在所有模式中支持 `renderExtraFooter`。[#13813](https://github.com/ant-design/ant-design/pull/13813) [@elfman](https://github.com/elfman
```
Comment (ycjcl868) [inline CHANGELOG.zh-CN.md]: 缺少一个 )

File: CHANGELOG.en-US.md [hunk 12]
```diff
@@ -15,6 +15,70 @@ timeline: true
...
+  - 🐞 Fixed rowSelection columnWidth doesn't work. [#14115](https://github.com/ant-design/ant-design/pull/14115) [@dyygtfx](https://github.com/dyygtfx)
+- DatePicker
+  - 🌟 Add new `renderFooter` API for DatePicker. [3c007a8](https://github.com/ant-design/ant-design/commit/3c007a85dd8a670f66c7e0aed95d3537e25ef6ea)
+  - 🐞 Fixed dateRender not supported at WeekPicker. [#13957](https://github.com/ant-design/ant-design/issues/13957)
+  - 🐞 Fixed disabled button style in DatePicker panel. [#14098](https://github.com/ant-design/ant-design/pull/14098)
+- Calendar
+  - 🌟 Calendar support prop `renderExtraFooter` in all mode. [#13813](https://github.com/ant-design/ant-design/pull/13813) [@elfman](https://github.com/elfman
+  - 🐞 Fixed month range display when start year equals end year. [#14049](https://github.com/ant-design/ant-design/pull/14049) [@meihuanyu](https://github.com/meihuanyu)
+- TimePicker
+  - 🌟 TimePicker support new prop `popupStyle` and `onAmPmChange`. [833c181](https://github.com/ant-design/ant-design/commit/833c18192247f265b4004afa11a054846c7ba662
```
Comment (ycjcl868) [inline CHANGELOG.en-US.md]: lack a `)`

File: CHANGELOG.en-US.md (hunk 10 above)
Comment (chenshuai2144) [reply CHANGELOG.en-US.md]: 虽然不知道怎么改，但是感觉这个英语怪怪的

File: CHANGELOG.zh-CN.md (hunk 7 above)
Comment (ycjcl868) [inline CHANGELOG.zh-CN.md]: 没有链接

File: CHANGELOG.en-US.md [hunk 13]
```diff
@@ -15,6 +15,70 @@ timeline: true
...
+  - 🐞 Fixed TimePicker Icon disappear when used with Input.Group. [#13797](https://github.com/ant-design/ant-design/pull/13797) [@mraiguo](https://github.com/ant-design/ant-design/pull/13797)
+- 🌟 Update `rc-tree-select` to 2.5.0 to support use a ReactNode as `notFoundContent` in TreeSelect. [47b89e5](https://github.com/ant-design/ant-design/commit/47b89e56fbedfa07a4c263ca390a78d58132563f)
+- 🌟 Rate component support `tooltips`. [192e188](https://github.com/ant-design/ant-design/commit/192e188fe87018ad9d004b9c2002e2b0045fa4b4)
+- 🐞 Fixed Drawer prop `style` not working. [#13850](https://github.com/ant-design/ant-design/issues/13850)
+- Upload
+  - 💄 Add new less var `upload-picture-card-border-style` and Fixed `upload-picture-card-size` typo. [#13919](https://github.com/ant-design/ant-design/pull/13919) [#13929](https://github.com/ant-design/ant-design/pull/13929) [@flexchen](https://github.com/flexchen)
+  - 🐞 Fixed recognize `dpg` file type as image in Upload. [#13988]
```
Comment (ycjcl868) [inline CHANGELOG.en-US.md]: lack PR link

**Comment by picodoth** [comment]
@chenshuai2144 @ycjcl868 改好了

File: CHANGELOG.zh-CN.md [hunk 14]
```diff
@@ -15,6 +15,69 @@ timeline: true
 
 ---
 
+## 3.12.0
+
+- 🎉 新的组件 [Empty](https://ant.design/components/empty/)！[13651](https://github.com/ant-design/ant-design/pull/13651)
+- 🎉 添加新的国际化资源（kn_IN）。[#13776](https://github.com/ant-design/ant-design/pull/13776) [@gurungrahul2](https://github.com/gurungrahul2)
+- 🌟 新增 Icon `eye-invisible`。[b8630b3](https://github.com/ant-design/ant-design/commit/b8630b34556e58af31f51fb42d1299725ddd8219)
+- 🌟 ConfigProvider 添加 `prefixCls`。[#13389](https://github.com/ant-design/ant-design/pull/13389)
```
Comment (ycjcl868) [inline CHANGELOG.zh-CN.md]: ConfigProvider 组件，更明确一些，感觉

Comment (picodoth) [reply CHANGELOG.zh-CN.md]: done

**Pull request merged**